vm/actor/tests/paych_actor_test.rs

```diff
@@ -0,0 +1,1192 @@
+        .call(
+            &PAYCH_ACTOR_CODE_ID.clone(),
+            Method::UpdateChannelState as u64,
+            &Serialized::serialize(ucp).unwrap(),
+        )
+        .is_ok());
+
+    rt.verify();
+    let exp_ls = LaneState {
+        id: 0,
+        redeemed: BigInt::from(9),
+        nonce: 1,
+    };
+    let exp_state = PState {
+        from: state.from,
+        to: state.to,
+        to_send: TokenAmount::from(9 as u64),
+        settling_at: state.settling_at,
+        min_settle_height: state.min_settle_height,
+        lane_states: vec![exp_ls],
+    };
+    verify_state(&mut rt, 1, exp_state);
+}
+
+#[test]
+fn redeem_voucher_correct_lane() {
+    let bs = MemoryDB::default();
+    let (mut rt, mut sv) = require_create_cannel_with_lanes(&bs, 3);
+    let state: PState = rt.get_state().unwrap();
+    let initial_amount = state.to_send;
+    sv.amount = BigInt::from(9);
+    sv.lane = 1;
+    let ls_to_update: &LaneState = &state.lane_states[1];
+    sv.nonce = ls_to_update.nonce + 1;
+    let payee_addr = Address::new_id(103);
+    let payer_addr = Address::new_id(102);
+
+    rt.set_caller(ACCOUNT_ACTOR_CODE_ID.clone(), payee_addr);
+    rt.expect_validate_caller_addr(&[state.from, state.to]);
+    rt.expect_verify_signature(
+        sv.clone().signature.unwrap(),
+        payer_addr,
+        to_vec(&sv).unwrap(),
+        ExitCode::Ok,
+    );
+
+    let ucp = UpdateChannelStateParams {
+        proof: vec![],
+        secret: vec![],
+        sv: sv.clone(),
+    };
+
+    assert!(rt
+        .call(
+            &PAYCH_ACTOR_CODE_ID.clone(),
+            Method::UpdateChannelState as u64,
+            &Serialized::serialize(ucp).unwrap()
+        )
+        .is_ok());
+    rt.verify();
+
+    let state: PState = rt.get_state().unwrap();
+    let ls_updated: &LaneState = &state.lane_states[1];
+    let big_delta =
+        &sv.amount - BigInt::from_signed_bytes_be(&ls_to_update.redeemed.to_signed_bytes_be());
+
+    let exp_send = big_delta + BigInt::from_signed_bytes_be(&initial_amount.to_radix_be(10));
+    assert_eq!(
+        exp_send,
+        BigInt::from_radix_be(Sign::Plus, &state.to_send.to_radix_be(10), 10).unwrap()
+    );
+    assert_eq!(sv.amount, ls_updated.redeemed);
+    assert_eq!(sv.nonce, ls_updated.nonce);
+}
+
+#[test]
+fn merge_success() {
+    let num_lanes = 3;
+    let bs = MemoryDB::default();
+    let (mut rt, mut sv) = require_create_cannel_with_lanes(&bs, num_lanes);
+    let state: PState = rt.get_state().unwrap();
+    let state_2: PState = rt.get_state().unwrap();
+
+    rt.set_caller(ACCOUNT_ACTOR_CODE_ID.clone(), state.from);
+    let merge_to: &LaneState = &state.lane_states[0];
+    let merge_from: &LaneState = &state.lane_states[1];
+    sv.lane = merge_to.id;
+    let merge_nonce = merge_to.nonce + 10;
+    let merges: Vec<Merge> = vec![Merge {
+        lane: merge_from.id,
+        nonce: merge_nonce,
+    }];
+
+    sv.merges = merges;
+    let payee_addr = Address::new_id(103);
```

Comment:
Can you create constants for each of these, so you're not just referencing a random address for each. (For example this one is the payee address)

Reply:
I can do it for the majority of the values but they arent consistent with the variable names and values.

In one they set paych_addr to Address::new_id(100) and in another its  Address::new_id(101)

Reply:
Best solution is to probably put them in diff mods should be in my next commit